### DISCUSSION TITLE
[bug]: Please consider an alternative to LiveBlocks due to security/privacy

### DISCUSSION THREAD

**Discussion by prologic** [tldraw/tldraw-v1#556] [General]
Please [see this comment](https://github.com/tldraw/tldraw/discussions/347#discussioncomment-3940326) on the "Self Hosting" thread (_which apparently no-one has really figured out how to run this thing yet?! 😳_) -- I thought I'd sit down tonight and try, but then going back over the thread I realised one stickling point.

-  [Build amazing real‑time collaborative products - Liveblocks](https://liveblocks.io/)

This is essentially a 3rd-party external (yet another) Cloud / SaaS service that tldraw depends on for some (quite useful) feature(s) IMO -- Namely real-time collaboration/viewing of a drawing.

There's a couple of problems with this however:

- This is no longer a "tiny drawing app" as you immediately depend on one or more 3rd-party Cloud/SaaS services. I _believe_ even AWS S3 is required somewhere?
- This is security and privacy concern and basically makes the app unsuitable for self-hosting (at all).

I urge the project maintainers and/or developers to reconsider alternatives. Surely a simple backend can be written using Web Sockets that achieves the same desired behaviour and features? 🤔 -- I understand (anda actively use) [HedgeDoc](https://hedgedoc.org/) has a similar set of features and requires no 3rd-party Cloud/SaaS services.

Thank you ! 🙏

**Comment by steveruizok** [comment]
Hey, solid feedback. 

You do not have to use Liveblocks with tldraw (the library can support lots of multiplayer backends) though we use it for tldraw.com for our multiplayer rooms. We also use S3 as a media backend for images and videos uploaded to multiplayer rooms.

While I understand that this makes the self-hosting story more complicated, multiplayer for an app like tldraw is complex; in particular the use of images and other assets, as well as other services that touch the backend.

We are exploring other options for mutiplayer (including our own), however it's not clear that there will ever be a self-hosted solution that includes multiplayer. Excalidraw, a similar project, also [does not support collaborative features](https://github.com/excalidraw/excalidraw#self-hosting) on its self-hosted version. (HedgeDoc relies on yjs, which is open source and great for text but not ideal for apps like this one.)

Self-hosting was never a top priority for this project, though if we can figure out how to make it work for tldraw then I'd be happy to support it.
